test/pkg/framework/grafana.go

```diff
@@ -48,13 +55,24 @@ func DeployGrafanaInstance(namespace, labelName, labelValue string) error {
 					MatchLabels: map[string]string{labelName: labelValue},
 				},
 			},
-			Compat: &grafanav1.GrafanaCompat{},
 		},
 	}
 	if err := kubernetes.ResourceC(kubeClient).Create(grafanaCR); err != nil {
 		return fmt.Errorf("Error while creating Grafana CR: %v ", err)
 	}
 
+	// Patch Grafana CR with image based on digest to allow image mirroring. Can be removed when Garafana image is moved out of docker.io registry.
+	if deployment, err := GetDeploymentWaiting(namespace, defaultGrafanaDeployment, 2); err != nil {
+		return fmt.Errorf("Error while waiting for Grafana Deployment to become available: %v ", err)
+	} else if imageShaTag, imageEntryPresent := grafanaImages[deployment.Spec.Template.Spec.Containers[0].Image]; imageEntryPresent {
+		err := patchGrafanaCR(grafanaCR.GetNamespace(), grafanaCR.GetName(), func(grafana *grafanav1.Grafana) {
+			grafana.Spec.BaseImage = imageShaTag
+		})
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
should we set a specific error in case we have no matching for the Grafana image ? That would then be a fail-fast one.

Reply:
good point, added